File: ProgLangChallenge/Java/VoterCandidate.java
import java.util.*; 
public class VoterCandidate
{
    public VoterCandidate()
    {
    }

    public List<List<String>> getNamesAndScores(List<String> voter, List<List<String>> candidates)
    {
        List<List<String>> nameScores = new ArrayList<List<String>>();

        for(int i = 0; i < candidates.size(); i++)
        {
            nameScores.add(compareVoterAndCandidate(voter,candidates.get(i)));
        }
        return nameScores; 
    }

    private List<String> compareVoterAndCandidate(List<String> voter, List<String> candidate) 
    {
        List<String> nameScore = new ArrayList<String>(); 
        int score = 0;
        for(int i = 1; i < candidate.size(); i++)
        {
            if(voter.get(i).equals("0") || candidate.get(i).equals("0"))
            {
                continue;
            }
            else if (voter.get(i).equals(candidate.get(i)))
            {
                score += 1;
            }
            else 
            {
                score -= 1; 
            }   
        }
        nameScore.add(candidate.get(0));
        nameScore.add(String.valueOf(score));
        return nameScore;
    }
    
    public int findHighestScore(List<List<String>> nameScores)
    {
        int highestScore = Integer.parseInt(nameScores.get(0).get(1)); 
        for(int i = 1; i < nameScores.size(); i++)
        {
            int score = Integer.parseInt(nameScores.get(i).get(1));
            if(highestScore < score)
            {
                highestScore = score; 
            } 
        }
        return highestScore; 
    }
    
    public List<String> getCandidatesWithHishestScore(List<List<String>> nameScores,int highestScore)
    {
        List<String> names = new ArrayList<String>(); 
        for(int i = 0; i < nameScores.size(); i++)
        {
            int score = Integer.parseInt(nameScores.get(i).get(1));
            if(highestScore == score)
            {
                names.add(nameScores.get(i).get(0));
            }
        }
        return names; 
    }
}

File: ProgLangChallenge/C++/VoterCandidate.cpp
#include <iostream> 
#include <vector> 
#include "VoterCandidate.h" 

using namespace std;

vector<vector<string>> VoterCandidate::
getNamesAndScores(vector<string> voter, vector<vector<string>> candidates)
    {
        vector<vector<string>> nameScores;

        for(int i = 0; i < candidates.size(); i++)
        {
            nameScores.push_back(compareVoterAndCandidate(voter,candidates[i]));
        }
        return nameScores; 
    }

vector<string> VoterCandidate::
compareVoterAndCandidate(vector<string> voter, vector<string> candidate) 
    {
        vector<string> nameScore; 
        int score = 0;
        for(int i = 1; i < candidate.size(); i++)
        {
            if(voter[i].compare("0")==0 || candidate[i].compare("0")==0)
            {
                continue;
            }
            else if (voter[i].compare(candidate[i])==0)
            {
                score += 1;
            }
            else 
            {
                score -= 1; 
            }   
        }
        nameScore.push_back(candidate[0]);
        nameScore.push_back(to_string(score));
        return nameScore;
    }
 
 int VoterCandidate::findHighestScore(vector<vector<string>> nameScores)
    {
        int highestScore =  atoi(nameScores[0][1].c_str());
        
        for(int i = 1; i < nameScores.size(); i++)
        {
            int score = atoi(nameScores[i][1].c_str());
            if(highestScore < score)
            {
                highestScore = score; 
            } 
        }
        return highestScore; 
    }
    
vector<string> VoterCandidate::
getCandidatesWithHishestScore(vector<vector<string>> nameScores,int highestScore)
    {
        vector<string> names; 
        for(int i = 0; i < nameScores.size(); i++)
        {
            int score = atoi(nameScores[i][1].c_str());
            if(highestScore == score)
            {
                names.push_back(nameScores[i][0]);
            }
        }
        return names; 
    }